src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,72 @@
+import { ExecRenderBaseCommand } from "../../rendering/react/ExecRenderBaseCommand.js";
+import { appInstallationArgs } from "../../lib/app/flags.js";
+import { Flags } from "@oclif/core";
+import { ReactNode } from "react";
+import { getAppFromUuid, getAppVersionFromUuid } from "../../lib/app/uuid.js";
+import { normalizeAppInstallationId } from "../../normalize_id.js";
+
+export class UpgradeApp extends ExecRenderBaseCommand<
+  typeof UpgradeApp,
+  boolean
+> {
+  static description = "Upgrade target app to target version";
+  static args = {
+    ...appInstallationArgs,
+  };
+  static flags = {
+    "installation-id": Flags.string({
+      description: "target app to trigger upgrade for",
+      required: false,
+    }),
+    "target-version": Flags.string({
+      description: "target version to upgrade target app to",
+      required: true,
+      default: "latest",
+    }),
+  };
+
+  protected async exec(): Promise<boolean> {
+    let appInstallationId: string = checkInstallationIdInput(
```

Comment:
There are predefined methods for getting the app installation id from args/flags/context; please just use the `withAppInstallationId` function, and drop the flag and the `checkInstallationIdInput` function.